libraries/AP_HAL_ChibiOS/hwdef/YJUAV_A6SE_H743/hwdef.dat

```diff
@@ -0,0 +1,241 @@
+PB5   TIM3_CH2  TIM3 PWM(9)  GPIO(58)
+PB0   TIM3_CH3  TIM3 PWM(10) GPIO(59)
+PB1   TIM3_CH4  TIM3 PWM(11) GPIO(60)
+
+# PWM output for buzzer
+PD14 TIM4_CH3 TIM4 GPIO(77) ALARM
+
+# RC input
+PI7 TIM8_CH3 TIM8 RCININT PULLUP LOW
+
+# Analog in
+PC4 BATT_CURRENT_SENS ADC1 SCALE(1)
+PF11 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+
+# Analog sys 5v
+PF12 VDD_5V_SENS ADC1 SCALE(2)
+
+# ADC3.3/ADC6.6
+PC5 SPARE1_ADC1 ADC1 SCALE(1)
+PC0 SPARE2_ADC1 ADC1 SCALE(2)
+
+PC1 RSSI_IN ADC1 SCALE(1)
+
+PC3 FMU_SERVORAIL_VCC_SENS ADC1 SCALE(3)
+
+# CAN bus
+PD0 CAN1_RX CAN1
+PD1 CAN1_TX CAN1
+
+PB12 CAN2_RX CAN2
+PB13 CAN2_TX CAN2
+
+# GPIOs
+PC13 HEATER_EN OUTPUT LOW GPIO(80)
+define HAL_HEATER_GPIO_PIN 80
+define HAL_HAVE_IMU_HEATER 1
+
+# define default battery setup
+define HAL_BATT_VOLT_PIN 2
+define HAL_BATT_CURR_PIN 4
+define HAL_BATT_VOLT_SCALE 21
+define HAL_BATT_CURR_SCALE 34
+define HAL_BATT_MONITOR_DEFAULT 4
+
+#analog rssi pin (also could be used as analog airspeed input)
+# PC1
+define BOARD_RSSI_ANA_PIN 11
+
+# enable pins
+PC14 VDD_3V3_SENSORS_EN OUTPUT LOW
+
+# red LED marked as B/E
+PE15 LED_R1 OUTPUT OPENDRAIN HIGH GPIO(0)
+PD10 LED_G1 OUTPUT OPENDRAIN LOW GPIO(1)
+PG0  LED_B1 OUTPUT OPENDRAIN HIGH GPIO(2)
+
+define HAL_GPIO_A_LED_PIN 0
+define HAL_GPIO_B_LED_PIN 1
+define HAL_GPIO_C_LED_PIN 2
+
+define HAL_GPIO_LED_ON  0
+define HAL_GPIO_LED_OFF 1
+
+# use pixracer style 3-LED indicators
+define HAL_HAVE_PIXRACER_LED
+
+# allow to have have a dedicated safety switch pin
+define HAL_HAVE_SAFETY_SWITCH 1
+PB15 LED_SAFETY OUTPUT
+PA4 SAFETY_IN INPUT PULLDOWN
+
+# SPI devices 
+SPIDEV imu1	        SPI1 DEVID1  IMU1_CS         MODE3  2*MHZ  8*MHZ
+SPIDEV imu2	        SPI6 DEVID1  IMU2_CS         MODE3  2*MHZ  8*MHZ
+SPIDEV ramtron      SPI5 DEVID1  FRAM_CS         MODE3  8*MHZ  8*MHZ
+SPIDEV dps310       SPI5 DEVID3  BAROMETER1_CS   MODE3   5*MHZ  5*MHZ
+SPIDEV ms5611       SPI5 DEVID3  BAROMETER2_CS   MODE3   5*MHZ  5*MHZ
+
+# IMU1
+IMU Invensense SPI:imu1 ROTATION_NONE
+IMU Invensensev3 SPI:imu1 ROTATION_NONE
+
+# IMU2
+IMU Invensense SPI:imu2 ROTATION_NONE
+IMU Invensensev3 SPI:imu2 ROTATION_NONE
+
+define HAL_DEFAULT_INS_FAST_SAMPLE 3
+
+# baro
+BARO DPS310 SPI:dps310
+BARO MS56XX SPI:ms5611
+BARO ICP201XX I2C:0:0x64
+
+# compass
+define HAL_COMPASS_DISABLE_IST8310_INTERNAL_PROBE
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+
+COMPASS IST8310 I2C:ALL_EXTERNAL:0x0E true  ROTATION_ROLL_180_YAW_90
+COMPASS IST8310 I2C:0:0x0E false ROTATION_YAW_180
+
+# microSD support
+PC8  SDMMC1_D0  SDMMC1
+PC9  SDMMC1_D1  SDMMC1
+PC10 SDMMC1_D2  SDMMC1
+PC11 SDMMC1_D3  SDMMC1
+PC12 SDMMC1_CK  SDMMC1
+PD2  SDMMC1_CMD SDMMC1
+
+# enable FAT filesystem support (needs a microSD defined via SDMMC)
+define HAL_OS_FATFS_IO 1
+
+define HAL_BOARD_LOG_DIRECTORY "/APM/LOGS"
+define HAL_BOARD_TERRAIN_DIRECTORY "/APM/TERRAIN"
+
+# enable RAMTROM parameter storage
+define HAL_STORAGE_SIZE 32768
+define HAL_WITH_RAMTRON 1
+
+DMA_PRIORITY SPI1*  TIM*UP*
+DMA_NOSHARE  SPI1*  TIM*UP*
```

Comment:
I would suggest no share for SPI6 as well

Reply:
This is a good suggestion.